internal/kafka/internal/workers/kafka_mgrs/promotion/promotion_kafkas_mgr.go

```diff
@@ -0,0 +1,97 @@
+package promotion
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/dbapi"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/config"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/services"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/workers/kafka_mgrs/promotion/chain"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/workers/kafka_mgrs/promotion/internal/actions"
+	errors2 "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/workers"
+	"github.com/golang/glog"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+)
+
+type PromotionKafkaManager struct {
+	workers.BaseWorker
+	kafkaService        services.KafkaService
+	quotaServiceFactory services.QuotaServiceFactory
+	kafkaConfig         *config.KafkaConfig
+}
+
+func NewPromotionKafkaManager(reconciler workers.Reconciler,
+	kafkaService services.KafkaService,
+	kafkaConfig *config.KafkaConfig,
+	quotaServiceFactory services.QuotaServiceFactory) *PromotionKafkaManager {
+	return &PromotionKafkaManager{
+		BaseWorker: workers.BaseWorker{
+			Id:         uuid.New().String(),
+			WorkerType: "promoting_kafka",
+			Reconciler: reconciler,
+		},
+		kafkaService:        kafkaService,
+		quotaServiceFactory: quotaServiceFactory,
+		kafkaConfig:         kafkaConfig,
+	}
+}
+
+// Start initializes the kafka manager to reconcile accepted kafka requests.
+func (k *PromotionKafkaManager) Start() {
+	k.StartWorker(k)
+}
+
+// Stop causes the process for reconciling accepted kafka requests to stop.
+func (k *PromotionKafkaManager) Stop() {
+	k.StopWorker(k)
+}
+
+func (k *PromotionKafkaManager) updatePromotionDetails(kafkaRequest *dbapi.KafkaRequest, promotionError error) error {
+	kafkaRequest.PromotionDetails = promotionError.Error()
+
+	if serviceError, ok := promotionError.(*errors2.ServiceError); ok && serviceError.Recoverable() {
+		kafkaRequest.PromotionStatus = dbapi.KafkaPromotionStatusPromoting
+	} else {
+		kafkaRequest.PromotionStatus = dbapi.KafkaPromotionStatusFailed
+	}
+
+	err := k.kafkaService.Update(kafkaRequest)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (k *PromotionKafkaManager) Reconcile() []error {
+	glog.Infoln("reconciling kafkas to be promoted")
+	kafkasToPromote, err := k.kafkaService.ListKafkasToBePromoted()
+	if err != nil {
+		return []error{errors.Wrap(err, "failed to list kafkas to promote")}
+	} else {
```

Comment:
No need for an `else`.
If there is no error we can unconditionally log as if there was an error we return before